src/cocotb/types/logic_array.py

```diff
@@ -859,13 +859,8 @@ def __setitem__(
                 raise IndexError(
                     f"slice [{start}:{stop}] direction does not match array direction [{self.left}:{self.right}]"
                 )
-            value_as_logics = [
-                Logic(v) for v in cast(Iterable[LogicConstructibleT], value)
-            ]
-            if len(value_as_logics) != (stop_i - start_i + 1):
-                raise ValueError(
-                    f"value of length {len(value_as_logics)!r} will not fit in slice [{start}:{stop}]"
-                )
+            value = cast(Union[str, Iterable[LogicConstructibleT], int], value)
+            value_as_logics = LogicArray(value, stop_i - start_i + 1)
```

Comment:
While this is less performant than the previous implementation, it is more flexible, and honestly a rare operation.